lib/mqttclient.ts

```diff
@@ -296,21 +406,19 @@ export default class MQTTClient {
         // Get the path as everything behind the last /
         let path = birth.name.substring(0, birth.name.lastIndexOf("/"));
 
-        // Here we only currently store the InstanceUUID and SchemaUUID
-        // of the top-level birth certificate. It would be nice if we could
-        // also store the InstanceUUID and SchemaUUID of all nested schemas
-        // but unsure how to handle this in InfluxDB.
-
         writeApi.useDefaultTags({
-            instance: birth.instance,
-            schema: birth.schema,
+            topLevelInstance: birth.instance.top,
+            bottomLevelInstance: birth.instance.bottom,
+            usesInstances: birth.instance.top + ':' + birth.instance.full.join(':'),
```

Comment:
Does InfluxDB make it straightforward to filter by a tag like this? If I am looking for a particular Instance_UUID, can I easily pull out those entries?

If this entry is supposed to be a set it would probably be better to sort it, so identical sets of UUIDs come out as identical strings. If this is supposed to be an ordered list of UUIDs going down the metric tree then the code needs to pull the UUIDs out in the correct order.

(I assume it's not possible to create a tag which is a list or a set, but we have to stringify...)

Reply:
> Does InfluxDB make it straightforward to filter by a tag like this? If I am looking for a particular Instance_UUID, can I easily pull out those entries?

Using regex is the advised way.

![image](https://github.com/AMRC-FactoryPlus/influxdb-sparkplug-ingester/assets/114239316/0dcf87b1-3a73-4656-a6bd-6b2fa1ae5177)

> I assume it's not possible to create a tag which is a list or a set, but we have to stringify...

Unfortunately not.